### DISCUSSION TITLE
Make it possible to diff between branches

### DISCUSSION THREAD

**Discussion by steinarb** [magit/magit#5053] [Ideas]
Today I wanted to diff a branch against a branch from a different remote that I had added and there didn't seem like a way to do this in magit? 

Places I looked were 'b' (expecting something like 'b d' to do a diff, but that was already taken by branch delete)
And then I looked in 'd' (expecting something like 'd b' and then be prompted for a branch to diff against).

Then I googled and didn't find anything.

I think this would be a nice feature to have, and even without knowing anything about magit I suspect it wouldn't be very hard to implement.

I don't want or need to do a fancy side-by-side diff. I just want what "git diff" would do when asked to compare against a different branch.

Thanks!

- Steinar

**Comment by jacobilsoe** [comment]
Hi

Have you tried d r (magit-diff-range)?

Best regards,
Jacob

**Comment by steinarb** [comment]
>>>>> Jacob Ilsø ***@***.***>:

> Have you tried d r (magit-diff-range)?

I thought that was something else, based on the description and name
("what's a 'range'...?" was my thought... still is...).

But giving a different branch than the one I was on gave me the diff
between that branch and my current one.

So thanks for that heads up! :-)

But the changes in the diff go the opposite way of the way I would like
to see the changes.

So the get the diffs the right way I have to be on the origin branch and
use the target branch as "the range"

Thanks!

**Reply by jacobilsoe** [reply]
Glad to hear it worked out. :-) You can read more here: https://magit.vc/manual/magit/Diffing.html#index-d-r
